### DISCUSSION TITLE
`steps.1.value` is empty even though the step shows data

### DISCUSSION THREAD

**Discussion by SomniVertix** [Budibase/budibase#7719] [Help] [answered]
When setting up the Automation tab, I have 3 sections: a) Hook into application action b) run javascript to do some data grooming c) call external data connector (HTTP POST Endpoint) with the data.

![image](https://user-images.githubusercontent.com/35609960/189497243-c8749028-c500-4d05-a9fd-ad2dc6884ac8.png)

However, when looking at the data, the message shows as {{ steps.1.value }} instead of the actual value

I imagine its simpel and I'm missing something minor but I'm not sure. the step value shows as filled in on the actual step

**Comment by melohagan** [comment, marked as answer]
Hey @SomniVertix 

The binding should work, it just doesn't show the actual binding value in the input dialog unfortunately. 
I noticed this myself and had raised an issue here: https://github.com/Budibase/budibase/issues/7543

Are you having issues with the External Data Connector then? What does the output say?

**Reply by SomniVertix** [reply]
Ah okay - So, it looks like I just had to do some extra steps on my end. 

For anyone looking at this after the fact:

I was trying to set up budibase to send **multiline data** to a REST endpoint. When budibase shows this in the step 1 `output` section, it escapes all of the new lines and such. However, when its being sent **to the REST endpoint, it sends the raw data unescaped**. Which lead to the request being malformed json.

The solution was to **escape the multiline data myself using json stringify** and then delete the beginning and end quotes.

For a more concrete/visual, here the example code:

```
const some_external_variable = "{{ trigger.fields.some_external_variable }}"

const index_templates = `
my ${some_external_variable}
multiline
data
`

let format = index_templates
format = JSON.stringify(format)
format = format.slice(1, -1)

return format;
```
